tensorboard/webapp/widgets/line_chart_v2/lib/chart.ts

```diff
@@ -0,0 +1,161 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {Chart, ChartCallbacks, ChartOption} from './chart_types';
+import {Coordinator} from './coordinator';
+import {
+  DataSeries,
+  DataSeriesMetadataMap,
+  Dimension,
+  Extent,
+} from './internal_types';
+import {
+  ObjectRenderer,
+  RendererType,
+  SvgRenderer,
+  ThreeRenderer,
+} from './renderer';
+import {createScale} from './scale';
+import {ScaleType} from './scale_types';
+import {SeriesLineView} from './series_line_view';
+import {ThreeCoordinator} from './threejs_coordinator';
+
+/**
+ * Workaround for testability.
+ *
+ * Inside TensorBoard unit test, we set up zone.js for Angular components and it patches
+ * asynchronousity. For unknown reasons, `fakeAsync` and `flush` misbehave and make test
+ * authoring more cumbersome.
+ */
+const util = {
+  requestAnimationFrame: self.requestAnimationFrame,
```

Comment:
`globalThis` ~ `self` ~ `window` (on the main thread; in case of Worker, you don't have `window` but only `globalThis` and `self`). Changed it to window for now but you will see reference to `self` soon :)

Reply:
Huh. TIL!